### ISSUE TITLE
Upgrading 10.9 to 10.9.1 fails to set default EnableArcMapRuntime true

### ISSUE THREAD

**Issue by BrendanEsri** [Esri/arcgis-powershell-dsc#400]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request

### Module Version

* 3.3.0

### Affected Resource(s)

* ArcGIS_Server

### Configuration Files

```
# Copy-paste your DSC JSON configurations here - for large configs,
# please use a service like Dropbox and share a link to the ZIP file.
```

### Expected Behavior
On upgrades to 10.9.1 that do not have the EnableArcMapRuntime set to false, have it automatically enabled

### Actual Behavior
On upgrade to 10.9.1, when EnableArcMapRuntime is not set it will not enable it. When you go through an InstallLicenseConfigure after the upgrade, it still will not install the ArcMap Runtime support when the parameters is not set (should be though due to the default value being true)

I ran this over in 3.3.2 on a 10.9.1 ArcGIS Server and it did not set it (did not have the EnableArcMapRuntime parameter set but it should set since default value is set to true)

### Steps to Reproduce

1. Install 10.9 ArcGIS Server without parameter EnableArcMapRuntime on 3.3.0
2. Upgrade 10.9 to 10.9.1 without parameter EnableArcMapRuntime on 3.3.0
3. Run InstallLicenseConfigure without parameter EnableArcMapRuntime on 3.3.2
Notice how the ArcGIS Server Program does not have that setting enabled:
 

### Important Factoids
I added another node to step 3 (it installed arcmap runtime on that new node, but failed to change the setting on the existing node

Received error:
Failed to Join Site after multiple attempts. Error on last attempt:- Failed to join the site '[https://existingmachine:6443/arcgis/admin'.](https://existingmachine:6443/arcgis/admin.) Failed to join the site '[https://existingmachine:6443/arcgis/admin'.](https://existingmachine:6443/arcgis/admin.) The state of the ArcMap Runtime Support feature configured on machine 'dotappq004vw.state.in.us' does not match site running at '[https://existingmachine:6443/arcgis/admin'.](https://existingmachine:6443/arcgis/admin.) All machines in a site are required to have the same ArcMap Runtime Support configuration. Information on how to resolve this issue is available in the help topic 'About multiple-machine ArcGIS Server sites'.

I ended up removing the runtime on the new node then running through powershell again and it successfully joined the machine to the site

### References
https://enterprise.arcgis.com/en/server/latest/administer/windows/disable-arcmap-runtime.htm

**Comment by cameronkroeker** [comment]
Hi @BrendanEsri,

I believe the reported issue here stems from two things. First, I believe step 2 was caused by a bug in v3.3.0, that was addressed in v3.3.1: https://github.com/Esri/arcgis-powershell-dsc/issues/359.

And the second item, opting in or out of ArcMapRuntime after the software has already been installed requires a modify install, which will need to be done manually through the Windows Add/Remove Program feature. The module supports opting out of ArcMapRuntime during a fresh installation of 10.9.1 or upgrade to 10.9.1. This is why you observed the original AGS node keeping ArcMap Runtime disabled, and the newly added node having it enabled.

Thanks,
Cameron K.

**Comment by spitzerr** [comment]
> And the second item, opting in or out of ArcMapRuntime after the software has already been installed requires a modify install, which will need to be done manually through the Windows Add/Remove Program feature. 

@cameronkroeker  Shouldn't/Couldn't this be implemented to make it more "desired state"?
 and 
"idempotent" in regards of reapplying configurations and changing configurations to change the system?

**Comment by cameronkroeker** [comment]
> > And the second item, opting in or out of ArcMapRuntime after the software has already been installed requires a modify install, which will need to be done manually through the Windows Add/Remove Program feature.
> 
> @cameronkroeker Shouldn't/Couldn't this be implemented to make it more "desired state"? and "idempotent" in regards of reapplying configurations and changing configurations to change the system?

I agree, adding support for modify install is a more desirable state. This feature request will be elevated for a future release.

Thanks,
Cameron K.
